### PR TITLE
Add DataPostprocessor flavor of MaterialModelInputs constructor.

### PR DESCRIPTION
**Pull request by ian-r-rose**
Factor out another big chunk of `MaterialModelInputs` initialization into a constructor.

### CODE REVIEW

File: include/aspect/material_model/interface.h [hunk 1]
```diff
@@ -184,12 +185,28 @@ namespace aspect
...
 
       /**
        * Constructor. Initializes the various arrays of this
        * structure with the FEValues and introspection objects and
        * the solution_vector. This constructor calls the function
        * reinit to populate the newly created arrays.
+       *
+       * @param input_data The data used to populate the material model inpu quantities.
+       * @param introspection A reference to the simulator introspection object.
+       * @param use_strain_rate Whether to compute the strain rates.
```
Comment (bangerth) [inline include/aspect/material_model/interface.h]: the first two lines here don't match the actual argument names

File: include/aspect/material_model/interface.h [hunk 2]
```diff
@@ -184,12 +185,28 @@ namespace aspect
       MaterialModelInputs(const unsigned int n_points,
                           const unsigned int n_comp);
 
+      /**
+       * Constructor. Initialize the arrays of the structure with the number
+       * of points in the `input_data` structure, and fills them appropriately.
+       *
+       * @param input_data The data used to populate the material model inpu quantities.
```
Comment (bangerth) [inline include/aspect/material_model/interface.h]: -> input

File: source/material_model/interface.cc [hunk 3]
```diff
@@ -240,6 +240,44 @@ namespace aspect
...
+      position(input_data.solution_values.size()),
+      temperature(input_data.solution_values.size(), numbers::signaling_nan<double>()),
+      pressure(input_data.solution_values.size(), numbers::signaling_nan<double>()),
+      pressure_gradient(input_data.solution_values.size(), numbers::signaling_nan<Tensor<1,dim> >()),
+      velocity(input_data.solution_values.size(), numbers::signaling_nan<Tensor<1,dim> >()),
+      composition(input_data.solution_values.size(), std::vector<double>(introspection.n_compositional_fields, numbers::signaling_nan<double>())),
+      strain_rate(input_data.solution_values.size(), numbers::signaling_nan<SymmetricTensor<2,dim> >()),
+      cell(NULL)
+    {
+      this->position = input_data.evaluation_points;
```
Comment (bangerth) [inline source/material_model/interface.cc]: you can actually do this assignment already in the initialization above (first line of member initialization)

**Comment by bangerth** [comment]
In the spirit of #1537.

File: include/aspect/material_model/interface.h (hunk 1 above)
Comment (ian-r-rose) [inline include/aspect/material_model/interface.h]: Oops, copy-paste error.

**Comment by ian-r-rose** [comment]
Comments addressed, all green.

File: include/aspect/material_model/interface.h (hunk 2 above)
Comment (bangerth) [inline include/aspect/material_model/interface.h]: You're still short on Ts.

**Comment by ian-r-rose** [comment]
Whoops

**Comment by ian-r-rose** [comment]
Okay, how about now?

**Comment by bangerth** [comment]
Nice!

**Pull request merged**